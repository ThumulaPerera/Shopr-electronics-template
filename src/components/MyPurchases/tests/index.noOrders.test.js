/* eslint-disable no-undef */
import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Header, Button } from 'semantic-ui-react';
import { MyPurchases as UnconnectedMyPurchases } from '../index';

jest.unmock('react-redux-firebase');
const rrf = require.requireActual('react-redux-firebase');

let component;

beforeEach(() => {
  // mock fn for isLoaded
  rrf.isLoaded = jest.fn(() => true);

  // eslint-disable-next-line no-console
  console.error = jest.fn(); /* disable proptype warnings */

  component = shallow(
    <UnconnectedMyPurchases
      auth={{ uid: '123' }}
      cart={[]}
      match={{
        params: {
          storeID: 'store123',
        },
      }}
    />,
  );
});

afterEach(cleanup);

it('renders purchases empty message if purchases are empty', () => {
  expect(component).toMatchSnapshot();

  // renders cart empty message
  const message = (
    <Header
      as="h2"
      style={{ fontSize: '1.3em', fontWeight: 'normal' }}
    >
      You don&apos;t have any purchases
    </Header>
  );
  expect(component).toContainReact(message);

  // renders start shopping button
  const btn = (
    <Button
      as="a"
      href="/store123"
      primary
      size="big"
    >
      Start shopping
    </Button>
  );
  expect(component).toContainReact(btn);
});
