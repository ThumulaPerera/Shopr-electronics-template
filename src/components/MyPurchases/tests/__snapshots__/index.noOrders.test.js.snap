// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders purchases empty message if purchases are empty 1`] = `
<Segment
  basic={true}
  style={
    Object {
      "marginTop": "5em",
    }
  }
  textAlign="center"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Purchases -
       
    </title>
  </HelmetWrapper>
  <Header
    as="h2"
    style={
      Object {
        "fontSize": "1.3em",
        "fontWeight": "normal",
      }
    }
  >
    You don't have any purchases
  </Header>
  <Divider
    hidden={true}
  />
  <Button
    as="a"
    href="/store123"
    primary={true}
    size="big"
  >
    Start shopping
  </Button>
</Segment>
`;
